daq/stream_monitor.py

```diff
@@ -141,11 +142,11 @@ def event_loop(self):
         try:
             if not fds and self.idle_handler:
                 self.idle_handler()
-                # Check corner case when idle_handler removes all callbacks.
-                if not self.callbacks:
-                    return False
             if self.loop_hook:
                 self.loop_hook()
+                # Check the corner case when loop_hook removes all callbacks.
```

Comment:
why is the idle_handler check going away?  Seems like if anything it would need to be in both places?

Reply:
Unless I’m missing something, I think only shutdown can remove all callbacks. Since the shutdown is moved out of idle handler to loop hook, I moved the check to be under loop hook.

Reply:
It sounds like you're using a-priori knowledge about some other class as applied to this class -- a dangerous game.  In theory, the idle_handle could remove all the callbacks, there's nothing that prevents it from doing so.  You're statement should be "currently only the shutdown method from some other class actually removed all callbacks" -- anything _can_ do it, which is different than what does it now.

Let me put it this way:  Is there something wrong with the check after idle_handler for this, or were you removing it because you knew it wasn't used?